library/Icingadb/View/BaseHostAndServiceRenderer.php

```diff
@@ -0,0 +1,263 @@
+    {
+        $title->addHtml(Html::sprintf(
+            $this->translate('%s is %s', '<hostname> is <state-text>'),
+            $this->createSubject($item, $layout),
+            Html::tag('span', ['class' => 'state-text'], $item->state->getStateTextTranslated())
+        ));
+
+        if (isset($item->state->affects_children) && $item->state->affects_children) {
+            $total = (int) $item->total_children;
+
+            if ($total > 1000) {
+                $total = '1000+';
+                $tooltip = $this->translate('Up to 1000+ affected objects');
+            } else {
+                $tooltip = sprintf(
+                    $this->translatePlural(
+                        '%d affected object',
+                        'Up to %d affected objects',
+                        $total
+                    ),
+                    $total
+                );
+            }
+
+            $icon = new Icon(Icons::UNREACHABLE);
+
+            $title->addHtml(new HtmlElement(
+                'span',
+                Attributes::create([
+                    'class' => 'affected-objects',
+                    'title' => $tooltip
+                ]),
+                $icon,
+                Text::create($total)
+            ));
+        }
+    }
+
+    public function assembleExtendedInfo($item, HtmlDocument $info, string $layout): void
+    {
+        if ($item->state->is_overdue) {
+            $since = new TimeSince($item->state->next_update->getTimestamp());
+            $since->prepend($this->translate('Overdue') . ' ');
+            $since->prependHtml(new Icon(Icons::WARNING));
+
+            $info->addHtml($since);
+        } elseif ($item->state->last_state_change !== null && $item->state->last_state_change->getTimestamp() > 0) {
+            $info->addHtml(new TimeSince($item->state->last_state_change->getTimestamp()));
+        }
+    }
+
+    public function assembleFooter($item, HtmlDocument $footer, string $layout): void
+    {
+        $pieChartLimit = 5;
+        $statusIcons = new HtmlElement('div', Attributes::create(['class' => 'status-icons']));
+        $isService = $item instanceof Service;
+
+        if (
+            ($isService && $item->state->last_comment->service_id === $item->id)
+            || $item->state->last_comment->host_id === $item->id
+        ) {
+            $comment = $item->state->last_comment;
+
+            if ($isService) {
+                $comment->service = $item;
+            } else {
+                $comment->host = $item;
+            }
+
+            $statusIcons->addHtml(
+                new HtmlElement(
+                    'div',
+                    Attributes::create(['class' => 'comment-wrapper']),
+                    new HtmlElement(
+                        'div',
+                        Attributes::create(['class' => 'comment-popup']),
+                        new ObjectHeader($comment) // TODO: popup shows object link
```

Comment:
No wonder. (referring to the todo) Don't use the object header. That is precisely why we separated rendering from usage! Use the item layout with the comment renderer directly and nothing else!